xchange-stream-btcmarkets/src/main/java/info/bitrich/xchangestream/btcmarkets/BTCMarketsStreamingService.java

```diff
@@ -3,28 +3,58 @@
 import com.fasterxml.jackson.databind.JsonNode;
 import com.google.common.collect.Lists;
 import com.google.common.collect.Sets;
-import info.bitrich.xchangestream.btcmarkets.dto.BTCMarketsWebSocketSubscribeMessage;
+import info.bitrich.xchangestream.btcmarkets.dto.BTCMarketsWebSocketSubscriptionMessage;
 import info.bitrich.xchangestream.service.netty.JsonNettyStreamingService;
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 class BTCMarketsStreamingService extends JsonNettyStreamingService {
+  //	TODO change to enumerator
   static final String CHANNEL_ORDERBOOK = "orderbook";
   static final String CHANNEL_HEARTBEAT = "heartbeat";
+  static final String CHANNEL_TICKER = "tick";
+  static final String CHANNEL_TRADE = "trade";
+
   private static final Logger LOG = LoggerFactory.getLogger(BTCMarketsStreamingService.class);
-  private final Set<String> subscribedOrderbooks = Sets.newConcurrentHashSet();
+  private final ConcurrentHashMap<String, Set<String>> subscribedMarketIds =
+      new ConcurrentHashMap<String, Set<String>>();
 
   public BTCMarketsStreamingService(String apiUrl) {
     super(apiUrl);
   }
 
-  private BTCMarketsWebSocketSubscribeMessage buildSubscribeMessage() {
-    return new BTCMarketsWebSocketSubscribeMessage(
-        new ArrayList<>(subscribedOrderbooks),
-        Lists.newArrayList(CHANNEL_ORDERBOOK, CHANNEL_HEARTBEAT),
+  /*
+   * Implementation renamed from BTCMarketsWebSocketSubscribeMessage to BTCMarketsWebSocketSubscriptionMessage to look after
+   * more than just the OrderBook subscription. This new implementation also incorporates the use of adding subscriptions to an existing one
+   * instead of having to resubscribing with all channels and all {@code marketIds} every time the data services calls the subscribe methods.
+   */
+  private BTCMarketsWebSocketSubscriptionMessage buildSubscribeMessage(
+      String channelName, Set<String> marketIds) {
+
+    // Create the first subscription message
+    if (!hasActiveSubscriptions()) {
+      return BTCMarketsWebSocketSubscriptionMessage.getFirstSubcritionMessage(
+          Lists.newArrayList(marketIds),
```

Comment:
can we avoid using google classes and use native new ArrayList<>(marketIds) instead?